api4/group.go

```diff
@@ -581,14 +581,15 @@ func getGroups(c *Context, w http.ResponseWriter, r *http.Request) {
 		channelID = id
 	}
 
-	if teamID == "" && channelID == "" && !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
-		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
-		return
-	}
+	// if teamID == "" && channelID == "" && !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+	// 	c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+	// 	return
+	// }
```

Comment:
@mkraft, what do you think should be the right permission here to be able to fetch the list of groups from the server (same q for PERMISSION_MANAGE_TEAM below)?

Reply:
wouldnt the new `PERMISSION_USE_GROUP_MENTIONS` be the ideal permission since anyone who is able to mention groups will need to be able to grab the list of groups from the API?

Reply:
@catalintomai when the endpoint originally shipped we had it to `manage_system`. Then a chat-facing feature required access for team and channel admins and it was opened up to all. Later it was deemed too permissive and a change was made to limit access to team/channel admins (effectively) via the `manage_team` and `manage_channel` permissions if a `team_id` or `channel_id` query parameter was present.

However, the new feature for at-mentioning groups now requires that we provide the specific ability to limit reference groups so that guests cannot view them. 

It's sounding to me like it needs it's own permission, something like `view_reference_groups` or perhaps `reference_groups` which combines at-mentioning them at seeing them?

I think non-reference groups could also have their own permission, but we could keep the same-old `manage_system`, `manage_team`, `manage_channel` permission check for now for MVP. I'm 0/5 on that but maybe PM should be looped in on that decision because it's not strictly necessary for the feature but does have a minor UX impact.

Reply:
@fm2munsh, @mkraft, maybe what we need is a permission that is more generic (smth. like view_groups), not tied to this particular scenario of group mentioning (or others), also a permission that guests would not have - still not clear to me if guests should be able to see/fetch groups at all. We can consider the allowreference flag on the group through which we can filter/hide the groups that we don't want referenced from both UX and mid-tier as orthogonal to this permission (meaning that the permission does not know/care about the referenceable status of the groups that it gates). Thoughts?

Reply:
I'm struggling to think of a use case for enforcing `view_reference_groups` and so I'm currently favouring @fm2munsh's suggestion of reusing `use_group_mentions`.

What would be a use case for turning off `view_groups`? Would you need both `view_groups` and `use_group_mentions` to mention a group or would you just not get the autosuggest list?

Reply:
I agree that `use_group_mentions` works ok today. I'm just a bit wary of conflating it with `view_groups`. What is tomorrow we introduce another scenario , let's call it `scenariox` that also requires permissions to fetch groups. Would we use the 1) `use_groups_mentions` permission to gate that as well (and same for scenarios y, z, etc.), which can be done but doesn't sound right. Or is it be better to have 2) smth. like `use_groups_scenariox` which would be used in smth like `if(use_group_mentions || use_groups_scenariox...) then you are ok.` or to have 3) a `view_groups` generic permission that gates the viewing of groups for all possible scenarios that need it and then have per-scenario permissions (like `use_groups_mentions`, `use_groups_scenariox`) that could be used  in other places, not only when fetching groups. 
Also, what happens if we want to gate surfacing groups to a category of people (e.g. guests). In approach #2 we would have to make sure that guests are not assigned roles that have permissions `use_groups_mentions` and `use_groups_scenariox`, while in approach #3 we can gate groups access for a persona for all scenarios .  OTOH I also don't like to have inflation of permissions, so adding a new permission without real actual justification today doesn't sound right either.